src/OAuth2.ts

```diff
@@ -0,0 +1,109 @@
+import * as querystring from "querystring";
+import * as rp from "request-promise-native";
+import * as uuid from "uuid/v4";
+import { Logging } from "matrix-appservice-bridge";
+
```

Comment:
why are we splitting imports all over the place? Bunch them all up together please.